models/authentication/authentication-exceptions.ts

```diff
@@ -0,0 +1,43 @@
+import { Exception } from '#models/exceptions';
+
+export class AgentConnectionFailedException extends Exception {
+  constructor(args: { cause?: any }) {
+    super({
+      name: 'AgentConnectionFailed',
+      ...args,
+    });
+  }
+}
+
+export class PrestataireException extends Exception {
+  constructor(args: { cause?: any }) {
+    super({
+      name: 'PrestataireException',
+      ...args,
+    });
+  }
+}
+
+export class CanRequestAuthorizationException extends Exception {
+  constructor(codeJuridique: string, siren: string) {
+    super({
+      name: 'OrganisationCanRequestAuthorizationException',
+      context: {
+        details: codeJuridique,
+        siren,
+      },
+    });
+  }
+}
+
+export class NoSiretException extends Exception {
```

Comment:
Is this still possible with the new ProConnect feature?

Reply:
In theory no, but we want to prove theory wrong